homeassistant/components/switchbot/coordinator.py

```diff
@@ -22,40 +26,94 @@ def flatten_sensors_data(sensor):
     return sensor
 
 
-class SwitchbotDataUpdateCoordinator(DataUpdateCoordinator):
+class SwitchbotCoordinator:
```

Comment:
If we split the entity details logic out from the passive bluetooth coordinator it looks like we can re-use much of that here. Inherit from it here and implement overrides as needed.

Reply:
Will refactor after https://github.com/home-assistant/core/pull/75642 merges

Reply:
I think we might need a more generic processor class for this case but I can refactor the existing one to subclass a newer more generic one

Reply:
I think we can separate the processor into PassiveBluetoothDataProcessor and PassiveBluetoothEntityDataProcessor

Reply:
Not so much reusable but 3 of the functions are so I'll make it a base class instead and then subclass it here to use it as a processor for switch bot

Reply:
I was just thinking we can reuse the coordinator, not the processor.

Reply:
I refactored so there are two coordinator options

`PassiveBluetoothDataUpdateCoordinator`, similar to the existing `DataUpdateCoordinator` but the caller needs to subclass `_async_handle_bluetooth_event` to handle the data

`PassiveBluetoothProcessorCoordinator`, what used to be called `PassiveBluetoothDataUpdateCoordinator` and is used for the `inkbird`/`xiaomi_ble`/`sensorpush` so far.

Reply:
I was thinking of moving all the processor based ones into `passive_update_processor.py`

Reply:
Looks good. I don't have an opinion about moving things.

Reply:
I'll move it around so I don't confused the two of them and then make a new PR